src/deserialize/parse.rs

```diff
@@ -907,6 +926,96 @@ mod test {
         )
     }
 
+    #[test]
+    fn parse_discard_valid() {
+        let mut edn = "#_iasdf 234".chars().enumerate();
+        let res = parse(edn.next(), &mut edn).unwrap();
+
+        assert_eq!(res, Edn::UInt(234))
+    }
+
+    #[test]
+    fn parse_discard_invalid() {
+        let mut edn = "#_{ 234".chars().enumerate();
+        let res = parse(edn.next(), &mut edn);
+
+        assert_eq!(
+            res,
+            Err(Error::ParseEdn(
+                "None could not be parsed at char count 3".to_string()
+            ))
+        )
+    }
+
+    #[test]
+    fn parse_discard_space_valid() {
+        let mut edn = "#_ ,, 234 567".chars().enumerate();
+        let res = parse(edn.next(), &mut edn).unwrap();
+
+        assert_eq!(res, Edn::UInt(567))
+    }
+
+    #[test]
+    fn parse_discard_space_invalid() {
+        let mut edn = "#_ ,, #{hello, this will be discarded} #_{so will this} #{this is invalid"
+            .chars()
+            .enumerate();
+        let res = parse(edn.next(), &mut edn);
+
+        assert_eq!(
+            res,
+            Err(Error::ParseEdn(
+                "None could not be parsed at char count 58".to_string()
+            ))
+        )
+    }
+
+    #[test]
+    fn parse_discard_empty() {
+        let mut edn = "#_ ,, foo".chars().enumerate();
+        let res = parse(edn.next(), &mut edn).unwrap();
+
+        assert_eq!(res, Edn::Empty)
+    }
+
+    #[test]
+    fn parse_discard_repeat_empty() {
+        let mut edn = "#_ ,, #_{discard again} #_ {:and :again} :okay"
+            .chars()
+            .enumerate();
+        let res = parse(edn.next(), &mut edn).unwrap();
+
+        assert_eq!(res, Edn::Empty)
+    }
+
+    #[test]
+    fn parse_discard_repeat_not_empty() {
+        let mut edn = "#_ ,, #_{discard again} #_ {:and :again} :okay {:a map}"
```

Comment:
this seems weird, parsing `:okay {:a map}` seems like a problem to return:
```
Edn::Map(Map::new(
  map! {":a".to_string() => Edn::Symbol("map".to_string())}
))
```

Did you check the standard response? I would expect the lib to return the keyword :okay and ignore the map

Reply:
yea let me check what standard lib does with that.

this is how it works here: 
```
(discard 
  (discard {discard again}
           return (discard {:and :again}
                           return :okay)))
  return {:a map})
```

Reply:
Here it is:
```test.clj
clj-scratch.core> (edn/read-string "#_ ,, #_{discard again} #_ {:and :again} :okay {:a map}")
{:a map}
clj-scratch.core> 
```

Reply:
Nice